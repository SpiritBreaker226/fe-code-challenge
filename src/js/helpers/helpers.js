import {screen} from '@testing-library/react';

export const elementToRender = testId => () => screen.queryByTestId(testId);
