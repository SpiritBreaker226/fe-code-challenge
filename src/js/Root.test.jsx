import React from 'react';

import {render, screen, waitFor} from '@testing-library/react';

import axios from 'axios';

import {elementToRender} from './helpers/helpers';
import spots from './helpers/test_data_spots.json';

import Root from './Root';

jest.mock('axios');
const mockedAxios = axios;

describe('Root', () => {
    describe('on buying a spot from search', () => {
        it('renders Root component', async () => {
            mockedAxios.get.mockResolvedValueOnce({data: spots});

            render(<Root />);

            await waitFor(elementToRender('spot-item-1'));

            expect(mockedAxios.get).toHaveBeenCalledTimes(1);
            expect(mockedAxios.get).toHaveBeenCalledWith('/spots');
        });
    });

    describe('when loading the page', () => {
        it('should render the loading text', async () => {
            mockedAxios.get.mockResolvedValueOnce({data: []});

            render(<Root />);

            expect(screen.getByText('Loading', {exact: false})).toBeInTheDocument();
            expect(screen.getByText('Loading', {exact: false})).toMatchSnapshot();
            expect(screen.queryByTestId('data-testid')).not.toBeInTheDocument();
        });
    });
});

