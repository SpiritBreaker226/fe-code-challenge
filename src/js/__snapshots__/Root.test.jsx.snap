// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root when loading the page should render the loading text 1`] = `
<div
  class="Root-loader"
>
  Loading...
</div>
`;
